Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -443,154 +452,158 @@ void FieldProbe::ComputeDiags (int step)
+        {
+            const auto getPosition = GetParticlePosition<FieldProbePIdx>(pti);
+            auto const np = pti.numParticles();
+            const auto& arrEx = Ex[pti].array();
+            const auto& arrEy = Ey[pti].array();
+            const auto& arrEz = Ez[pti].array();
+            const auto& arrBx = Bx[pti].array();
+            const auto& arrBy = By[pti].array();
+            const auto& arrBz = Bz[pti].array();
+
+            /*
+             * Make the box cell centered in preparation for the interpolation (and to avoid
+             * including ghost cells in the calculation)
+             */
+            amrex::Box box = pti.tilebox();
+            box.grow(Ex.nGrowVect());
+
+            // preparing to write data to particle
+            auto& attribs = pti.GetStructOfArrays().GetRealData();
+            ParticleReal* const AMREX_RESTRICT part_Ex = attribs[FieldProbePIdx::Ex].dataPtr();
+            ParticleReal* const AMREX_RESTRICT part_Ey = attribs[FieldProbePIdx::Ey].dataPtr();
+            ParticleReal* const AMREX_RESTRICT part_Ez = attribs[FieldProbePIdx::Ez].dataPtr();
+            ParticleReal* const AMREX_RESTRICT part_Bx = attribs[FieldProbePIdx::Bx].dataPtr();
+            ParticleReal* const AMREX_RESTRICT part_By = attribs[FieldProbePIdx::By].dataPtr();
+            ParticleReal* const AMREX_RESTRICT part_Bz = attribs[FieldProbePIdx::Bz].dataPtr();
+            ParticleReal* const AMREX_RESTRICT part_S = attribs[FieldProbePIdx::S].dataPtr();
+
+            auto* const AMREX_RESTRICT idcpu = pti.GetStructOfArrays().GetIdCPUData().data();
+
+            const amrex::XDim3 xyzmin = WarpX::LowerCorner(box, lev, 0._rt);
+            const amrex::XDim3 dinv = WarpX::InvCellSize(lev);
+            const Dim3 lo = lbound(box);
+
+            // Temporarily defining modes and interp outside ParallelFor to avoid GPU compilation errors.
+            const int temp_modes = WarpX::n_rz_azimuthal_modes;
+            const int temp_interp_order = interp_order;
+            const bool temp_field_probe_integrate = m_field_probe_integrate;
+
+            // Interpolating to the probe positions for each particle
+            amrex::ParallelFor(np, [=] AMREX_GPU_DEVICE(long ip)
             {
-                const auto &arrEx = Ex[pti].array();
-                const auto &arrEy = Ey[pti].array();
-                const auto &arrEz = Ez[pti].array();
-                const auto &arrBx = Bx[pti].array();
-                const auto &arrBy = By[pti].array();
-                const auto &arrBz = Bz[pti].array();
+                amrex::ParticleReal xp, yp, zp;
+                getPosition(ip, xp, yp, zp);
+
+                amrex::ParticleReal Exp = 0._prt, Eyp = 0._prt, Ezp = 0._prt;
+                amrex::ParticleReal Bxp = 0._prt, Byp = 0._prt, Bzp = 0._prt;
+
+                // Amrex particle container should guarantee that all particles are in domain
+                AMREX_ASSERT(inDomainChecker(xp, yp, zp));
```

Comment:
This assert will only be executed in debug mode.

Do you mean to write an `if` here?

```suggestion
                if (inDomainChecker(xp, yp, zp))
                {
                    // doGatherShapeN
                    // calc poynting vector
                }
```
and assign/add the default zeros in `else` to the particle.

Reply:
When I did testing, I wasn't able to create a situation where there was a particle in the container that was outside of the problem domain. To improve my understanding, let me know if there is a straightforward way to create that situation, I can try to add a test that creates that situation.

If the particle is not in the problem domain, I thought we could not call doGatherShapeN (or at least wouldn't get reliable results). If `inDomainChecker(...)` is false, what values should I use for E field, B fields, and S vector?